### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by Tulleb** [AgentOps-AI/agentops#308]
## 🐛 Bug Report

**🔎 Describe the Bug**
I can't initiate a session working fine on Python AgentOps `0.2.6` but not on `0.3.X` (at the moment on 0.3.2).

**🔄 Reproduction Steps**
Calling `agentops.init()` with a properly setup `AGENTOPS_API_KEY`.
The terminal then display the following error:
```
🖇 AgentOps: Error: No session exists

2024-07-21 11:39:16,858 - 6119288832 - meta_client.py-meta_client:55 - WARNING: Error: No session exists
🖇 AgentOps: Error: list index out of range
2024-07-21 11:39:16,858 - 6119288832 - meta_client.py-meta_client:55 - WARNING: Error: list index out of range

2024-07-21 11:39:16,858 - 6119288832 - app.py-app:838 - ERROR: Exception on /__/functions.yaml [GET]
Traceback (most recent call last):
  File "/Users/.../functions-py/venv/lib/python3.12/site-packages/agentops/meta_client.py", line 53, in wrapper
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../functions-py/venv/lib/python3.12/site-packages/agentops/client.py", line 566, in _safe_get_session
    raise NoSessionException("No session exists")
agentops.exceptions.NoSessionException: No session exists

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/.../functions-py/venv/lib/python3.12/site-packages/agentops/meta_client.py", line 53, in wrapper
    return method(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../functions-py/venv/lib/python3.12/site-packages/agentops/client.py", line 455, in create_agent
    session = self._safe_get_session()
              ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/.../functions-py/venv/lib/python3.12/site-packages/agentops/meta_client.py", line 61, in wrapper
    self._sessions[0],  # TODO: find which session caused exception
    ~~~~~~~~~~~~~~^^^
IndexError: list index out of range
```

**🙁 Expected Behavior**
Session should start.

**🔍 Additional Context**
Same code is working fine when downgrading to AgentOps 0.2.6.

Here is my `main.py` file initiating AgentOps:
```
"""
Main module for Firebase function with logging, environment validation,
and agent operations initialization.
""
import logging
import os
from firebase_functions import (
  https_fn,
  options,
  storage_fn,
)
from firebase_admin import (
  initialize_app,
  firestore,
)
from ai.crew import crew
import agentops

initialize_app()

# Configure logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

# Validate required environment variables
REQUIRED_ENV_VARS = [
  'OPENAI_API_KEY',
  'SERPER_API_KEY',
  'AGENTOPS_API_KEY',
  'OPENAI_MODEL_NAME',
]

for var in REQUIRED_ENV_VARS:
  if not os.getenv(var):
    logger.error('Error: %s environment variable is not set.', var)
    raise EnvironmentError(f"{var} environment variable is not set.")

agent_ops_api_key = os.getenv('AGENTOPS_API_KEY')

agentops.init(
  api_key=agent_ops_api_key,
  auto_start_session=False,
)
logger.info('Agentops initialized successfully.')

@storage_fn.on_object_finalized(
  timeout_sec=600,
  memory=options.MemoryOption.GB_1,
  region="europe-west1"
)
def onGuideUpload(event: storage_fn.CloudEvent[storage_fn.StorageObjectData]):
  """When a guide is uploaded in the Storage bucket, inject a new AI generated guide in the database."""

  logger.info('Function triggered.')

  # Exit if this is triggered on a file that is not a guide.
  guide_id = event.data.name
  if not guide_id or not guide_id.startswith("guide/"):
    print(f"This is not a guide: {guide_id}")
    return

  guide_id = guide_id.replace("guide/", "")

  try:
    guide_url = 'https://someURL.pdf'
    logger.info('Guide ID: %s', guide_id)
    logger.info('Guide URL: %s', guide_url)

    agentops.start_session()

    result = crew.kickoff(
      inputs={
        "guide_id": guide_id,
        "guide_url": guide_url,
        # "guide_document": document
      }
    )
    logger.info('Crew kickoff completed.')
    logger.info('Result: %s', result)

  except Exception as e:
    agentops.end_session('Fail')
    logger.error('Error during crew.kickoff: %s', e)

    generation_failed(guide_id=guide_id)

    return https_fn.Response("Server Error: Failed during crew kickoff", status=500)

  agentops.end_session('Success')
  logger.info('Agentops session ended successfully.')

  return https_fn.Response(result, status=200)

def generation_failed(guide_id: str):
  """Update a guide from Firestore with the given informations and return its reference ID."""
  db = firestore.client()

  guide_data = {
    'isGenerated': True,
    'hasFailed': True,
  }

  doc_ref = db.collection('guides').document(guide_id)
  doc_ref.set(
    document_data=guide_data,
    merge=True,
  )

  return doc_ref.id
```

Here is the code at the `/Users/.../functions-py/ai/agents.py:32`:
```
reader = Agent(
  role='Guide Content Strategist',
  goal='Parse the guide into a JSON format',
  backstory="""
  Speaks JSON language fluently.
  Very used to read documents related to a house guide: explaining how a house is working, should be used, and what are the places to visit around it.
  These guides can be received into all kind of format, including pictures, PDFs, DOCXs, text files, etc.
  Strictly respect the given JSON format to fill.
  Always making sure no information is lost during the parsing process.
  Can use delegation to the Google Map Finder and the Instagram Finder to fill the missing informations.
  """,
  verbose=True,
  allow_delegation=True,
  # max_iter=10,
  tools=[scrape_tool, directory_tool, file_tool, pdf_tool, docx_tool],
)
```

Thank you for your great work!

**Comment by areibman** [comment]
Hey @Tulleb -- 

Can you upgrade to the latest release of AgentOps and try again?

**Comment by Tulleb** [comment]
Looks fixed in `0.3.4`, thank you guys!
